Fix layout of <ul> in messages

jackrzhang opened the pull request:
See #2671.

Before:
![image](https://user-images.githubusercontent.com/12771126/41494529-ebe1fc62-70c8-11e8-9234-af1d72a7b898.png)

Updated:

![image](https://user-images.githubusercontent.com/12771126/41494523-d75017ac-70c8-11e8-8dcf-394fbd266185.png)

@gnprice

gnprice commented:
Thanks for the swift fix!

I just merged the first commit, adding left padding.

For the second: it looks like you remove the bottom padding, but I don't see discussion of that in the commit message. I tried a quick test message, and it looks unbalanced to me and a bit cramped:

![image](https://user-images.githubusercontent.com/28173/41494800-12f65712-70ce-11e8-89fb-1f6c5d1d8506.png)

jackrzhang commented:
Updated:

<img src="https://user-images.githubusercontent.com/12771126/41497315-c4275462-7107-11e8-8300-8ae766122735.png" width=500>

I also added another commit so that the CSS rule-sets are correctly applied only to lists within messages.

gnprice commented:
LGTM, merged, thanks!

FTR, here's a screenshot of that same example message with the new code:
![image](https://user-images.githubusercontent.com/28173/41569403-fe5d6a6c-731f-11e8-9a6f-a977563607eb.png)
